native/templates/openharmony/entry/src/main/ets/MainAbility/cocos/jsb-adapter/sys-ability-polyfill.js

```diff
@@ -0,0 +1,25 @@
+import display from '@ohos.display';
+
+window.oh = {};
+
+module.exports = function systemReady () {
+    return new Promise(resolve => {
+        // HACK: strange, can't share the same global variable jsb
+        if (typeof jsb !== 'undefined') {
+            let device = jsb.device = {};
+            device.getDevicePixelRatio = function () { return 1; }
+            device.getDeviceOrientation = function () { return 0; }
+        }
+
+        if (typeof XMLHttpRequest === 'undefined') {
+            window.XMLHttpRequest = function () {}
+        }
+
+        display.getDefaultDisplay((err, data) => {
+            window.oh.display = data;
+            console.log('pptest display' +  data.width.toString()  + data.height.toString())
+        });
```

Comment:
```suggestion
        if (typeof XMLHttpRequest === 'undefined') {
            window.XMLHttpRequest = function () {}
        }

        display.getDefaultDisplay((err, data) => {
            window.oh.display = data;
            
            
            device.getDevicePixelRatio = function () { return data.densityPixels; }
            device.getDeviceOrientation = function () { return data.rotation; }
        });
```

I'm not sure wether this API works, please refer to the docs https://developer.harmonyos.com/cn/docs/documentation/doc-references/js-apis-display-0000001281001106